### PR TITLE
Add SettingsCache and SettingsObserver

### PR DESCRIPTION
**Pull request by agunde406**
Signed-off-by: Andrea Gunderson <agunde@bitwise.io>

### CODE REVIEW

File: validator/sawtooth_validator/execution/executor.py [hunk 1]
```diff
@@ -39,6 +40,10 @@
 LOGGER = logging.getLogger(__name__)
 
 
+TpSetting = namedtuple('TpSetting',
```
Comment (peterschwarz) [inline validator/sawtooth_validator/execution/executor.py]: Is this just used for storing one setting?  If so, it should probably be named for that particular setting

Comment (agunde406) [reply validator/sawtooth_validator/execution/executor.py]: Okay, I was following the format of `self._tp_settings_key = "sawtooth.validator.transaction_families`. Should it be called TransactionFamilies?

Comment (peterschwarz) [reply validator/sawtooth_validator/execution/executor.py]: That sounds like a good name

File: families/settings/tests/test_tp_settings.py [hunk 2]
```diff
@@ -56,6 +56,14 @@ def _expect_set(self, key, expected_value):
         self.validator.respond(
             self.factory.create_set_response(key), received)
 
+    def _expect_add_event(self, key):
+        recieved = self.validator.expect(
```
Comment (jsmitchell) [inline families/settings/tests/test_tp_settings.py]: s/recieved/received/

**Pull request merged**